engine/src/flutter/shell/platform/darwin/ios/framework/Headers/FlutterSceneLifeCycle.h

```diff
@@ -96,6 +98,45 @@ API_AVAILABLE(ios(13.0))
 
 @end
 
+/**
+ * A protocol for manually registering a `FlutterEngine` to receive scene life cycle events.
+ */
+@protocol FlutterSceneLifeCycleEngineRegistration
+/**
+ * Registers a `FlutterEngine` to receive scene life cycle events.
+ *
+ * This method is **only** necessary when the following conditions are true:
+ * 1. Multiple Scenes (UIApplicationSupportsMultipleScenes) is enabled.
+ * 2. The `UIWindowSceneDelegate` `window.rootViewController` is not a `FlutterViewController`
+ *    initialized with the target `FlutterEngine`.
+ *
+ * When multiple scenes is enabled (UIApplicationSupportsMultipleScenes), Flutter cannot
+ * automatically associate a `FlutterEngine` with a scene during the scene connection phase. In
+ * order for plugins to receive launch connection information, the `FlutterEngine` must be manually
+ * registered with either the `FlutterSceneDelegate` or `FlutterPluginSceneLifeCycleDelegate` during
+ * `scene:willConnectToSession:options:`.
+ *
+ * In all other cases, or once the `FlutterViewController.view` associated with the `FlutterEngine`
```

Comment:
nit: the `or` case seems a bit misleading as it sounded like Flutter automatically handle registration once `view` is added to the hierarchy even for manually managed engines?